File: photo_gallary/android/app/src/main/kotlin/com/photogallary/cheq/photo_gallary/mediastore/model/Photo.kt
package com.photogallary.cheq.photo_gallary.mediastore.model


data class Photo (
    val id : Long,
    val uri: String,
    val name: String,
    val size: Long,   // File size in bytes
    val timestamp: Long? = null,
    val path:String,
    val width : Int,
    val height : Int,
    val orientation: Int
){
    // Manually implement toJson() method to convert Media to Map<String, dynamic>
    fun toJson(): Map<String, Any?> {
        return mapOf(
            "id" to id,
            "uri" to uri,
            "name" to name,
            "size" to size,
            "timestamp" to timestamp,
            "path" to path,
            "width" to width,
            "height" to height,
            "orientation" to orientation
        )
    }
}

File: photo_gallary/lib/app/data/datasources/local/local_storage/model/photo.dart

class Photo {
      final int id;
      final String uri;
      final String name;
      final String path;
      final int size; // File size in bytes
      final int? timestamp; // Nullable timestamp
      final int width; // Width of the photo
      final int height; // Height of the photo

      Photo({
        required this.id,
        required this.uri,
        required this.name,
        required this.path,
        required this.size,
        this.timestamp,
        required this.width,
        required this.height,
      });

      // From JSON factory for converting JSON to Photo object
      factory Photo.fromJson(dynamic json) {
        return Photo(
          id: json['id'],
          uri: json['uri'],
          name: json['name'],
          path: json['path'],
          size: json['size'],
          timestamp: json['timestamp'],
          width: json['width'],
          height: json['height'],
        );
      }
}